pep-0722.rst

```diff
@@ -74,42 +73,41 @@ commonly-used approach.
 Specification
 =============
 
-Any Python script may contain one or more *metadata blocks*. Metadata blocks are
-identified by reading the script *as a text file* (i.e., the file is not parsed
-as Python source code), looking for contiguous blocks of lines that start with
-the identifying characters ``##``. Whitespace is not allowed before the
-identifying ``##``. More than one metadata block may exist in a Python file.
+The content of this section will be published in the Python Packaging user
+guide, PyPA Specifications section, as a document with the title "Embedding
+Metadata in Script Files".
 
-Tools reading metadata blocks MAY respect the standard Python encoding
-declaration. If they choose not to do so, they MUST process the file as UTF-8.
+Any Python script may contain a *dependency block*. The dependency block is
+identified by reading the script *as a text file* (i.e., the file is not parsed
+as Python source code), looking for the first line of the form::
 
-Within a metadata block, whitespace after the ``##`` and at the end of the line
-is ignored, and blank lines are ignored. The first line of a metadata block is
-special, and identifies the type of block. This line MUST contain a colon
-character, ``:``. If the colon is not present, the block is not considered to be
-a metadata block, and tools MUST ignore it. The block type is all of the text on
-the initial line, up to the colon. There must be no whitespace before the colon.
-The initial line MAY contain text after the colon. How this is interpreted
-depends on the block type. Block types MUST be treated as case sensitive.
+   # Script Dependencies:
 
-The interpretation of any lines in a metadata block after the initial
-identifying line, is defined by the type of block.
+The hash character must be at the start of the line with no preceding whitespace.
+The text "Script Dependencies" is recognised regardless of case, and the spaces
+represent arbitrary whitespace (although at least one space must be present). The
+following regular expression recognises the dependency block header line::
 
-Tools MUST ignore any blocks with types they do not handle.
+    (?i)^#\s+script\s+dependencies:\s*$
 
-Block types starting with the characters ``X-`` are reserved for the user, and
-MUST NOT be given a meaning in any future standard.
+Tools reading the dependency block MAY respect the standard Python encoding
+declaration. If they choose not to do so, they MUST process the file as UTF-8.
 
-Otherwise, the only defined block type is ``Script Dependencies``. For this
-block type,
+After the header line, all lines in the file up to the first line that doesn't
+start with a ``#`` sign are considered *dependency lines* and are treated as
+follows:
```

Comment:
I tried various ways and they all felt slightly clumsy in one way or another. Do you have a specific wording for that sentence in mind? If I just take what you wrote, I get

> After the header line, the block of consecutive lines starting with `#` are considered *dependency lines* and are treated as follows:

But this is grammatically incorrect, because it starts talking about the block, and ends talking about the individual lines. Any variation I tried was a mess, because I had to talk about a block, then say "split it into lines" and only *then* could I introduce the term "dependency line" and say what to do with the individual lines.